### PR TITLE
conf: scylla.yaml: update documentation for tablets

### LINKED ISSUE #19456: Update tablet documentation in scylla.yaml
Tablets are still listed in the commented experimental features section. Even though enabling this feature doesn't seem to do anything, it's still a little bit confusing:
https://github.com/scylladb/scylladb/blob/794440eb85b4da9f959a82de280d27836f489878/conf/scylla.yaml#L273-L279

Also, the new tablet feature description says:
https://github.com/scylladb/scylladb/blob/794440eb85b4da9f959a82de280d27836f489878/conf/scylla.yaml#L615-L616

I find this confusing, as I understood it as the option to control the default behavior when creating keyspace without the `tablets = {'enabled': true/false}` option specified. So that it would be possible to have tablets enabled in the cluster (so being able to create tablet keyspace), but still have vnode as the default replication type. 
On the other hand, it seems like it's working exactly like the previous experimental feature - it controls whether it's even possible to create tablet keyspace (and if it is possible, it makes it the default).

cc: @annastuchlik

### PR DESCRIPTION
**Pull request by bhalevy**
Tablets are no longer in experimental_features since 83d491a, so remove them from the experimental_features section documentation.

Also, expand the documentation for the `enable_tablets` option.

Fixes #19456

Needs backport to 6.0

### CODE REVIEW

File: conf/scylla.yaml [hunk 1]
```diff
@@ -612,5 +611,17 @@ maintenance_socket: ignore
...
+# When enabled, newly created keyspaces will have tablets enabled by default.
+# That can be explicitly disabled in the CREATE KEYSPACE query
+# by using the `tablets = {'enabled': false}` replication option.
+#
+# When the tablets feature is disabled, there is now way to enable tablets
+# per keyspace.
+#
+# Note that creating keyspaces with tablets-enabled is irreversible.
+# Existing keyspaces that were previously created with tablets
+# might have limited access if the tablets feature is later disabled
```
[the remark this answers is not in the document]
Comment (bhalevy) [reply conf/scylla.yaml:624]: okay.  that's vague enough :)

**Comment by bhalevy** [comment]
- fixed wording.

@annastuchlik please re-review

**Review by Michal-Leszczynski — APPROVED** [review]

**Comment by bhalevy** [comment]
@xemul please review/merge

**Comment by bhalevy** [comment]
@scylladb/scylla-maint please merge before branch-6.1 is forked

**Comment by bhalevy** [comment]
requires backport to 6.0, and then to 2024.2 when it reaches enterprise.

**Pull request closed without merging**